exporters/etw/README.md

```diff
@@ -0,0 +1,201 @@
+
+## Debugging
+
+### ETW TraceLogging Dynamic Events
+
+ETW supports a mode that is called "Dynamic Manifest", where event may contain strongly-typed
+key-value pairs, with primitive types such as `integer`, `double`, `string`, etc. This mode
+is optional and requires `TraceLoggingDynamic.h` header.
+
+Complete [list of ETW types](https://docs.microsoft.com/en-us/windows/win32/wes/eventmanifestschema-outputtype-complextype#remarks).
+
+OpenTelemetry C++ ETW exporter implements the following type mapping:
+
+| OpenTelemetry C++ API type | ETW type        |
+|----------------------------|-----------------|
+| bool                       | xs:byte         |
+| int (32-bit)               | xs:int          |
+| int (64-bit)               | xs:long         |
+| uint (32-bit)              | xs:unsignedInt  |
+| uint (64-bit)              | xs:unsignedLong |
+| double                     | xs:double       |
+| string                     | win:Utf8        |
+
+Support for arrays of primitive types is not implemented yet.
+
+Visual Studio 2019 allows to use `View -> Other Windows -> Diagnostic Events` to capture
+events that are emitted by instrumented application and sent to ETW provider in a live view.
+Instrumentation name passed to `GetTracer` API above corresponds to `ETW Provider Name`.
+If Instrumentation name contains a GUID - starts with a curly brace, e.g.
+`{deadbeef-fade-dead-c0de-cafebabefeed}` then the parameter is assumed to be
+`ETW Provider GUID`.
+
+Click on `Settings` and add the provider to monitor either by its Name or by GUID. In above
+example, the provider name is `OpenTelemetry-ETW-Provider`. Please refer to Diagnostic Events
+usage instructions [here](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-diagnostics-how-to-monitor-and-diagnose-services-locally#view-service-fabric-system-events-in-visual-studio)
+to learn more. Note that running ETW Listener in Visual Studio requires Elevation, i.e.
+Visual Studio would prompt you to confirm that you accept to run the ETW Listener process as
+Administrator. This is a limitation of ETW Listeners, they must be run as privileged process.
+
+### ETW events encoded in MessagePack
+
+OpenTelemetry ETW exporter optionally allows to encode the incoming event payload using
+[MessagePack](https://msgpack.org/index.html) compact binary protocol. ETW/MsgPack encoding
+requires [nlohmann/json](https://github.com/nlohmann/json) library to be included in the build
+of OpenTelemetry ETW exporter. Any recent version of `nlohmann/json` is compatible with ETW
+exporter. For example, the version included in `third_party/nlohmann-json` directory may be
+used.
+
+There is currently **no built-in decoder available** for this format. However, there is ongoing
+effort to include the ETW/MsgPack encoder in [Azure/diagnostics-eventflow](https://github.com/Azure/diagnostics-eventflow)
```

Comment:
Is it "include the ETW/MsgPack decoder"?

Reply:
My bad, yes :)